### PR TITLE
Simple Options

### PR DESCRIPTION
**Pull request by fifikim**
This PR meets the acceptance criteria for the Simple Options ticket:
- OPTIONS request to /method_options returns 200 status code and allowed headers of GET, HEAD, and OPTIONS
- OPTIONS request to /method_options2 returns 200 status code and allowed headers of GET, HEAD, OPTIONS, PUT and POST

### CODE REVIEW

File: src/main/java/server/router/routes/MethodOptions.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import server.constants.Method;
+import server.request.Request;
+import server.response.Response;
+import server.response.ResponseBuilder;
+
+public class MethodOptions implements RouteHandler {
+  @Override
+  public Set<Method> getMethods() {
+    return new LinkedHashSet<>(List.of(
+            Method.GET, Method.HEAD, Method.OPTIONS));
```
Comment (seoulection) [inline src/main/java/server/router/routes/MethodOptions.java:15]: Should we allow users to execute GET and HEAD requests to this endpoint? Similar question for `MethodOptions2`

Comment (fifikim) [reply src/main/java/server/router/routes/MethodOptions.java:15]: Users are able to execute all of the requests allowed for each of the endpoints and will receive a "HTTP/1.1 200 OK" response. If they send an Options request they'll also receive an Allow header. Should I be sending something different?

Comment (seoulection) [reply src/main/java/server/router/routes/MethodOptions.java:15]: I should have read the RFC first before sending my comment. Ignore me!

Comment (fifikim) [reply src/main/java/server/router/routes/MethodOptions.java:15]: Ok! Just making sure I wasn't overlooking something again!

**Review by seoulection — APPROVED** [review]

**Pull request merged**